docs/resources/tls_activation.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: "fastly"
+page_title: "Fastly: tls_activation"
+sidebar_current: "docs-fastly-resource-tls_activation"
+description: |-
+Enables TLS on a domain
+---
+
+# fastly_tls_activation
+
+Enables TLS on a domain using a specified custom TLS certificate.
+
+~> **Note:** The Fastly service must be provisioned _prior_ to enabling TLS on it. This can be achieved in Terraform using [`depends_on`](https://www.terraform.io/docs/configuration/meta-arguments/depends_on.html).
+
+## Example Usage
+
+Basic usage:
+
+```hcl
+resource "fastly_service_v1" "demo" {
+  name = "my-service"
+
+  domain {
+    name    = "example.com"
+  }
+
+  backend {
+    address = "127.0.0.1"
+    name    = "localhost"
+  }
+
+  force_destroy = true
+}
+
+resource "fastly_tls_private_key" "demo" {
+  key_pem = "..."
+  name = "demo-key"
+}
+
+resource "fastly_tls_certificate" "demo" {
+  certificate_body = "..."
+  name = "demo-cert"
+  depends_on = [fastly_tls_private_key.demo]
+}
+
+resource "fastly_tls_activation" "test" {
+  certificate_id = fastly_tls_certificate.demo.id
+  domain = "example.com"
```

Comment:
I'm not sure this would work, because `fastly_service_v1.demo.domain` refers to a set, not a list, and so can't be indexed directly. Alternatively `tolist(fastly_service_v1.test.domain)[0].name` could be used, or `[for d in fastly_service_v1.domain : d. name][0]` etc. Whichever you think is clearest between these or hardcoding 🙂

Reply:
OK good to know. I asked only for my own learning of the terraform hcl syntax (of which I'm still very green on). 👍🏻